ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+    /*      Field translation from GDAL to MiraMon                          */
+    /* -------------------------------------------------------------------- */
+    // Reset the object where readed coordinates are going to be stored
+    MMResetFeatureGeometry(&hMMFeature);
+    if (bcalculateRecord)
+    {
+        MMResetFeatureRecord(&hMMFeature);
+        if (!phMiraMonLayer->pLayerDB)
+        {
+            eErr = TranslateFieldsToMM();
+            if (eErr != OGRERR_NONE)
+                return eErr;
+        }
+        // Content field translation from GDAL to MiraMon
+        eErr = TranslateFieldsValuesToMM(poFeature);
+        if (eErr != OGRERR_NONE)
+        {
+            CPLDebug("MiraMon", "Error in MMProcessGeometry()");
+            return eErr;
+        }
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Write Geometry                                                  */
+    /* -------------------------------------------------------------------- */
+
+    // Reads objects with coordinates and transform them to MiraMon
+    if (poGeom)
+    {
+        eErr = MMLoadGeometry(OGRGeometry::ToHandle(poGeom));
+    }
+    else
+    {
+        if (!phMiraMonLayer->bIsBeenInit)
+        {
+            phMiraMonLayer->bIsDBF = TRUE;
+            MMInitLayerByType(phMiraMonLayer);
+            phMiraMonLayer->bIsBeenInit = 1;
+        }
+    }
+
+    // Writes coordinates to the disk
+    if (eErr == OGRERR_NONE)
+        return MMWriteGeometry();
+    CPLDebug("MiraMon", "Error in MMProcessGeometry()");
+    return eErr;
+}
+
+/****************************************************************************/
+/*                           ICreateFeature()                               */
+/****************************************************************************/
+
+OGRErr OGRMiraMonLayer::ICreateFeature(OGRFeature *poFeature)
+
+{
+    OGRErr eErr = OGRERR_NONE;
+
+    if (!bUpdate)
+    {
+        CPLError(CE_Failure, CPLE_NoWriteAccess,
+                 "\nCannot create features on a read-only dataset.");
+        return OGRERR_FAILURE;
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Write out the feature                                           */
+    /* -------------------------------------------------------------------- */
+    OGRGeometry *poGeom = LOG_ACTION(poFeature->GetGeometryRef());
+
+    // Processing a feature without geometry.
+    if (poGeom == nullptr)
+    {
+        eErr = LOG_ACTION(MMProcessGeometry(nullptr, poFeature, TRUE));
+        if (phMiraMonLayer->bIsDBF)
+            poFeature->SetFID(phMiraMonLayer->TopHeader.nElemCount - 1);
+        return eErr;
+    }
+
+    // At this point MiraMon does not support unkwnon type geometry
+    if (poGeom->getGeometryType() == wkbUnknown)
+        return LOG_ACTION(OGRERR_UNSUPPORTED_GEOMETRY_TYPE);
+
```

Comment:
that can't happen. only a layer/geometry field definition can have a unknown type. A OGRGeometry itself has necessary a known type
```suggestion
```

Reply:
Done